app_dart/lib/src/request_handlers/refresh_cirrus_status.dart

```diff
@@ -37,60 +42,91 @@ class RefreshCirrusStatus extends ApiRequestHandler<Body> {
     Config config,
     AuthenticationProvider authenticationProvider, {
     @visibleForTesting DatastoreServiceProvider datastoreProvider,
+    @visibleForTesting
+        this.branchHttpClientProvider = Providers.freshHttpClient,
+    @visibleForTesting this.gitHubBackoffCalculator = twoSecondLinearBackoff,
   })  : datastoreProvider =
             datastoreProvider ?? DatastoreService.defaultProvider,
+        assert(branchHttpClientProvider != null),
+        assert(gitHubBackoffCalculator != null),
         super(config: config, authenticationProvider: authenticationProvider);
 
   final DatastoreServiceProvider datastoreProvider;
+  final HttpClientProvider branchHttpClientProvider;
+  final GitHubBackoffCalculator gitHubBackoffCalculator;
 
   @override
   Future<Body> get() async {
     final DatastoreService datastore = datastoreProvider();
     final GraphQLClient client = await config.createCirrusGraphQLClient();
+    final GitHub github = await config.createGitHubClient();
+    const RepositorySlug slug = RepositorySlug('flutter', 'flutter');
+    final Stream<Branch> branches = github.repositories.listBranches(slug);
+    final List<String> regExps = await loadBranchRegExps(
+        branchHttpClientProvider, log, gitHubBackoffCalculator);
 
-    await for (FullTask task
-        in datastore.queryRecentTasks(taskName: 'cirrus', commitLimit: 15)) {
-      final String sha = task.commit.sha;
-      final String existingTaskStatus = task.task.status;
-      log.debug(
-          'Found Cirrus task for commit $sha with existing status $existingTaskStatus');
-      final List<String> statuses = <String>[];
-      const String name = 'flutter';
-
-      for (Map<String, dynamic> runStatus
-          in await queryCirrusGraphQL(sha, client, log, name)) {
-        final String status = runStatus['status'] as String;
-        final String taskName = runStatus['name'] as String;
-        log.debug('Found Cirrus build status for $sha: $taskName ($status)');
-        statuses.add(status);
+    await for (Branch branch in branches) {
+      if (!regExps
+          .any((String regExp) => RegExp(regExp).hasMatch(branch.name))) {
+        continue;
       }
+      await for (FullTask task in datastore.queryRecentTasks(
+          taskName: 'cirrus', commitLimit: 15, branch: branch.name)) {
```

Comment:
Should we add a constant for this limit? it can make it easier to change in the future.

Reply:
Added.